packages/orchestration/README.md

```diff
@@ -17,16 +17,143 @@ $ npm install @sap-ai-sdk/orchestration
+  llm: {
+    model_name: 'gpt-4-32k',
+    model_params: { max_tokens: 50, temperature: 0.1 }
+  },
+  templating: {
+    template: [
+      { role: 'user', content: 'What is the capital of {{?country}}?' }
+    ]
+  }
+});
+
+const response = await orchestrationClient.chatCompletion({
+  inputParams: { country: 'France' }
+});
+
+const responseContent = response.getContent();
+```
 
-This project is open to feature requests/suggestions, bug reports etc. via [GitHub issues](https://github.com/SAP/ai-sdk-js/issues).
+It is possible to provide a history of a conversation to the model.
+Use the following snippet to send a chat completion request with history and a system message:
+
+```TS
+import { OrchestrationClient } from '@sap-ai-sdk/orchestration';
+
+const orchestrationClient = new OrchestrationClient({
+  llm: {
+    model_name: 'gpt-4-32k',
+    model_params: { max_tokens: 50, temperature: 0.1 }
+  },
+  templating: {
+    template: [
+      { role: 'user', content: 'What is my name?' }
+    ],
+    messages_history: [
+      {
+        role: 'system',
+        content: 'You are a helpful assistant who remembers all details the user shares with you.'
+      },
+      {
+        role: 'user',
+        content: 'Hi! I am Bob'
+      },
+      {
+        role: 'assistant',
+        content: 'Hi Bob, nice to meet you! I am an AI assistant. I will remember that your name is Bob as we continue our conversation.'
+      }
+    ]
+  }
+});
+
+const responseContent = await orchestrationClient.chatCompletion().getContent();
+const tokenUsage = response.getTokenUsage();
+
+logger.info(
+  `Total tokens consumed by the request: ${tokenUsage.total_tokens}\n` +
+  `Input prompt tokens consumed: ${tokenUsage.prompt_tokens}\n` +
+  `Output text completion tokens consumed: ${tokenUsage.completion_tokens}\n`
+);
+```
+
+#### Content Filtering
+
+Use the orchestration client with filtering to restrict the type of content that is passed to and received from a generative AI model.
+
+This feature enables you always filter both input and the output of a model based on content safety criteria.
+
+```TS
+import { OrchestrationClient } from '@sap-ai-sdk/orchestration';
+
+const filter = buildAzureContentFilter({ Hate: 2, Violence: 4 });
+const orchestrationClient = new OrchestrationClient({
+  llm: {
+    model_name: 'gpt-4-32k',
+    model_params: { max_tokens: 50, temperature: 0.1 }
+  },
+  templating: {
+    template: [{ role: 'user', content: '{{?input}}' }]
+  },
+  filtering: {
+    input: filter,
+    output: filter
+  }
+});
+
+try {
+  const response = await orchestrationClient.chatCompletion({
+    inputParams: { input: 'I hate you!' }
+  });
+  return response.getContent();
+} catch (error: any) {
+  return `Error: ${error.message}`;
+}
+```
+
+The Azure content filter supports four categories: `Hate`, `Violence`, `Sexual`, and `SelfHarm`. Each category can be configured with severity levels of 0, 2, 4, or 6.
```

Comment:
[q] Should we document this in such detail here? I would imagine the jsdoc of the convenience function and generated types provide this info?

Reply:
As we don't have API docs yet, I would still like to mention it here.